h/static/scripts/group-forms/components/test/CreateGroupForm-test.js

```diff
@@ -3,20 +3,83 @@ import { mount } from 'enzyme';
 import CreateGroupForm from '../CreateGroupForm';
 
 describe('CreateGroupForm', () => {
-  const createWrapper = () => mount(<CreateGroupForm />);
+  /* Return true if counterEl is in its error state. */
+  const counterIsInErrorState = counterEl => {
+    return (
+      counterEl.hasClass('text-red-error') && counterEl.hasClass('font-bold')
+    );
+  };
+
+  /* Return true if component is in its error state. */
+  const componentIsInErrorState = component => {
+    return component.prop('error') !== undefined;
+  };
+
+  const getElements = wrapper => {
+    return {
+      name: {
+        counterEl: wrapper.find('[data-testid="charcounter-name"]'),
+        fieldComponent: wrapper.find('Input[data-testid="name"]'),
+        fieldEl: wrapper.find('input[data-testid="name"]'),
+      },
+      description: {
+        counterEl: wrapper.find('[data-testid="charcounter-description"]'),
+        fieldComponent: wrapper.find('Textarea[data-testid="description"]'),
+        fieldEl: wrapper.find('textarea[data-testid="description"]'),
+      },
+    };
+  };
+
+  const createWrapper = () => {
+    const wrapper = mount(<CreateGroupForm />);
+    const elements = getElements(wrapper);
+    return { wrapper, elements };
+  };
 
   [
-    { counter: 'name', limit: 25 },
-    { counter: 'description', limit: 250 },
-  ].forEach(({ counter, limit }) => {
-    describe(`${counter} character counter`, () => {
-      it(`initially renders 0/${limit}`, () => {
-        const wrapper = createWrapper();
-        const counterEl = wrapper.find(
-          `[data-testid="character-counter-${counter}"]`,
-        );
+    {
+      field: 'name',
+      limit: 25,
+    },
+    {
+      field: 'description',
+      limit: 250,
+    },
+  ].forEach(({ field, limit }) => {
+    describe(`${field} character counter`, () => {
+      it('renders a character counter', () => {
+        const { counterEl, fieldComponent } = createWrapper().elements[field];
 
         assert.equal(counterEl.text(), `0/${limit}`);
+        assert.isNotOk(counterIsInErrorState(counterEl));
+        assert.isNotOk(componentIsInErrorState(fieldComponent));
+      });
+
+      it("changes the count when the field's text changes", () => {
+        const { wrapper, elements } = createWrapper();
+        const fieldEl = elements[field].fieldEl;
+
+        fieldEl.getDOMNode().value = 'new text';
+        fieldEl.simulate('input');
+        wrapper.update();
```

Comment:
The `wrapper.update` call is not needed here as `simulate` implicitly triggers an update. Manual `update` calls are mainly needed when an update occurs as a result of something other than an Enzyme method call (eg. a timer firing or some external DOM event), where Enzyme is not aware of the change.

Reply:
Done